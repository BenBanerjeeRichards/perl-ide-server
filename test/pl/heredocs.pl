<<"";
   Hello!

    print <<OUT;
    OUT
    some OUT
OUT

print <<EOL;
    XHello World!
    Y qw {
EOL


print << "    OUT" unless @_[0];
    Hello. Output
    OUT


print <<~Hello;
Hello world!
Hello


print <<`EOL`;
    XHello World!
    Y qw {
EOL
