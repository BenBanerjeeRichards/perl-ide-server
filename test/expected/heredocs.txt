Operator(<<)
StringStart(")
String
StringEnd(")
;
HereDoc(   Hello!\n)
HereDocEnd
Builtin(print)
Operator(<<)
Name(OUT)
;
HereDoc(    OUT\n    some OUT\n)
HereDocEnd(OUT)
Builtin(print)
Operator(<<)
Name(EOL)
;
HereDoc(    XHello World!\n    Y qw {\n)
HereDocEnd(EOL)
Builtin(print)
Operator(<<)
StringStart(")
String(    OUT)
StringEnd(")
Unless
ArrayVariable(@_)
LSquareBracket
NumericLiteral(0)
RSquareBracket
;
HereDoc(    Hello. Output\n)
HereDocEnd(    OUT)
Builtin(print)
Operator(<<)
Operator(~)
Name(Hello)
;
HereDoc(Hello world!\n)
HereDocEnd(Hello)
Builtin(print)
Operator(<<)
StringStart(`)
String(EOL)
StringEnd(`)
;
HereDoc(    XHello World!\n    Y qw {\n)
HereDocEnd(EOL)
EndOfInput
